### PR TITLE
52143 show order detail

### PR DESCRIPTION
**Pull request by haltn-3346**
## Related Tickets
- [#TicketID](https://edu-redmine.sun-asterisk.vn/issues/52143)

## WHAT (optional)
- show history order of customer

## Evidence (Screenshot or Video)
image
image

### CODE REVIEW

File: app/views/order/_order.html.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<div class="tracking-item">
+  <div class="tracking-icon status-intransit"></div>
+  <div class="tracking-date">
+    <p class="badge btn <%= Settings.send(order.status) %>">
+      <%= t "status_order.#{order.status}" %>
+    </p>
+  </div>
+  <div class="tracking-content">
+    <span><b><%= order.created_at.strftime(Settings.date_format)  %></b></span>
```
Comment (trongxt-0611) [inline app/views/order/_order.html.erb:9]:
```suggestion
    <span><b><%= order.created_at.strftime(Settings.date_format) %></b></span>
```

File: app/views/customer/orders/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                      <td><%= number_to_currency(detail.price*detail.quality, locale: :vi) %></td>
+                    </tr>
+                  <% end %>
+                </tbody>
+              </table>
+            </div>
+          </div>
+          <div class="d-flex flex-row-reverse bg-light text-dark p-4">
+            <div class="py-3 px-5 text-right">
+              <div class="h2 font-weight-light"><%= t ".total" %><%=  total_price(@order)%></div>
```
Comment (trongxt-0611) [inline app/views/customer/orders/show.html.erb]:
```suggestion
              <div class="h2 font-weight-light"><%= t ".total" %><%=  total_price(@order) %></div>
```

File: app/views/customer/orders/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                    <th class="border-0 text-uppercase small font-weight-bold"><%= t ".name_product" %></th>
+                    <th class="border-0 text-uppercase small font-weight-bold"><%= t ".price" %></th>
+                    <th class="border-0 text-uppercase small font-weight-bold"><%= t ".quantity" %></th>
+                    <th class="border-0 text-uppercase small font-weight-bold"><%= t ".total_price" %></th>
+                  </tr>
+                </thead>
+                <tbody>
+                  <% @order_details.each do |detail| %>
+                    <tr>
+                      <td><%= image_tag detail.product.images[0] , style:"weight:80px; height:80px;"%></td>
```
Comment (trongxt-0611) [inline app/views/customer/orders/show.html.erb]:
```suggestion
                      <td><%= image_tag detail.product.images[0] , style:"weight:80px; height:80px;" %></td>
```

File: app/controllers/orders_controller.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  def show
+    @pagy, @order_details = pagy(@order.order_details.includes(:product),
+                                 items: Settings.pagy.items)
+    render "customer/orders/show"
+  end
+
+  private
+
+  def load_order
+    @order = current_user.orders.find_by(id: params[:id])
```
Comment (trongxt-0611) [inline app/controllers/orders_controller.rb]:
```suggestion
    @order = current_user.orders.find_by id: params[:id]
```

**Review by trongxt-0611 — COMMENTED** [review]
cmt

**Comment by haltn-3346** [comment]
ready

File: app/controllers/orders_controller.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+class OrdersController < ApplicationController
+  before_action :logged_in_user
+  before_action :load_order, except: :index
+  def index
```
Comment (ngocvt-0484) [inline app/controllers/orders_controller.rb:5]: thêm phía trên dòng trống

File: app/controllers/orders_controller.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    @pagy, @orders = pagy(current_user.orders.all.newest,
```
Comment (ngocvt-0484) [inline app/controllers/orders_controller.rb]:
```suggestion
    @pagy, @orders = pagy(current_user.orders.newest,
```

File: app/views/customer/orders/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,77 @@
+
```
Comment (ngocvt-0484) [inline app/views/customer/orders/show.html.erb]: bỏ dòng trống

**Comment by haltn-3346** [comment]
ready

**Pull request merged**